### PR TITLE
feat: patch for Bool lemmas from std4#183 and std4#329

### PR DESCRIPTION
**Pull request by fgdorais**
---

- [x] depends on: #8008
- [x] depends on: [std4#329](https://github.com/leanprover/std4/pull/329)

Built on top of #8008 which should be merged first.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Bool/Basic.lean [hunk 1]
```diff
@@ -270,17 +269,22 @@ attribute [simp] not_or
...
-  lt_iff_le_not_le _ _ := Bool.lt_iff_le_not_le
+  lt_iff_le_not_le := by decide
+  max_def := by decide
+  min_def := by decide
 #align bool.linear_order Bool.linearOrder
 
-attribute [simp] Bool.max_eq_or Bool.min_eq_and
+@[simp] theorem max_eq_or' : max = or := rfl
+
+@[simp] theorem min_eq_and' : min = and := rfl
```
Comment (eric-wieser) [inline Mathlib/Data/Bool/Basic.lean:284]: Should the Std lemmas be changed to match these ones instead?

Comment (fgdorais) [reply Mathlib/Data/Bool/Basic.lean:284]: OK.

Comment (eric-wieser) [reply Mathlib/Data/Bool/Basic.lean:284]: (if the answer at the Std end is no, then thanks for adding these back; I don't think they should have been deleted in https://github.com/leanprover-community/mathlib4/pull/7982/files#r1375519160)

Comment (fgdorais) [reply Mathlib/Data/Bool/Basic.lean:284]: I made them match and also simp in std4#329. The simp might be debatable in Std, I'm on the fence. Let's see what happens in Std and then fix here.

**Comment by eric-wieser** [comment]
It looks like you accidentally did a nonsensical rebase in the wrong direction; @semorrison has done this before by accident, and can perhaps share the secret for avoiding it.

**Comment by fgdorais** [comment]
> It looks like you accidentally did a nonsensical rebase in the wrong direction; @semorrison has done this before by accident, and can perhaps share the secret for avoiding it.

Thanks for the clue @eric-wieser! Assistance from you, @semorrison, or others is welcome. I'm currently fixing a bunch of these patches so I can't deal with deeper issues at this fast pace.

**Pull request closed without merging**